### DISCUSSION TITLE
keyExtractor item is sometimes undefined after item delete

### DISCUSSION THREAD

**Discussion by mgscreativa** [Shopify/flash-list#560] [Q&A]
Hi! using "@shopify/flash-list": "1.1.0" in an RN Expo (SDK 46) project. I have some issues with the keyExtractor (need it for prepareForLayoutAnimationRender() as stated here https://shopify.github.io/flash-list/docs/guides/layout-animation). I'm using "zustand": "^4.0.0" for state management, the items array comes from that store.

If I remove an item from the list (calling a zustand remove action), the keyExtractor sometimes get an undefined item to process.

**First of all, this is my keyExtractor**
```
keyExtractor={(item: NotificationItem, index: number) => {
  console.log('keyExtractor', item);
  // return index.toString();  <-- this works allways
  return item.msgId; <-- this fails for some renders after item deleted
}}
```
**This is my remove item hook**
```
const removeNotificationItem = (item: NotificationItem) => {
  removeNotification(item); <-- zustand remove action

  // Animate item deletion
  notificationsList.current?.prepareForLayoutAnimationRender();
  LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
};
```
**This is my item type definition**
```
export type NotificationItem = {
  msgId: string;
  title: string;
  message: string;
  link: string;
  read: boolean;
};
```
**EDIT:** It also happens in the renderItem prop. Have to add **_item && (_** to the renderItem function return because after item deletion, some items to render are undefined!
```
  const renderNotificationItem = ({ item }: { item: NotificationItem }) => {
    return (
      item && ( <-- Have to add this to avoid trying to render undefined items after deletion
        <Pressable
          onPress={() => {
            removeNotificationItem(item);
          }}>
          <View style={styles.listItem}>
            <Text>Cell Id: {item.title}</Text>
          </View>
        </Pressable>
      )
    );
  };
```

**This is the initial render with three items**
![image](https://user-images.githubusercontent.com/1286632/184360316-4a7cf513-4771-46ae-abcc-7a64720d6e7e.png)

And this are some errors I've got after one item deletion using my keyExtractor **return item.msgId;**
![image](https://user-images.githubusercontent.com/1286632/184360489-dfe05901-43a4-4546-8cc6-c7009aae146c.png)
![image](https://user-images.githubusercontent.com/1286632/184360551-2f28b0a1-f8ca-4f1e-9bcb-7acc47286597.png)
![image](https://user-images.githubusercontent.com/1286632/184360588-82acfeda-11d0-4a26-ba49-71959e7762aa.png)
![image](https://user-images.githubusercontent.com/1286632/184360634-d7a4b4b2-3ef4-4dd9-bec3-dca98fbca763.png)

**Comment by naqvitalha** [comment]
Will it be possible for you to create an expo snack for this so that I can take a look?
